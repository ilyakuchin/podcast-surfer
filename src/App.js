import React, { Component } from "react";
import "./App.css";
import Podcast from "./components/Podcast/Podcast";
import EpisodePlayer from "./components/EpisodePlayer/EpisodePlayer";
import { Switch, Route } from "react-router-dom";
import SearchPodcast from "./components/SearchPodcast/SearchPodcast";

export default class App extends Component {
  constructor(props) {
    super(props);

    this.state = {
      episodeInfo: null
    };
  }
  render() {
    return (
      <main>
        <Switch>
          <Route path="/" component={SearchPodcast} exact />
          <Route path="/podcast" component={Podcast} />
          <Route path="/episode-player" component={EpisodePlayer} />
        </Switch>
      </main>
    );
  }
}
