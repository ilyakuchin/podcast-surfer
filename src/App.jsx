import React from 'react';
import { Switch, Router } from 'react-router-dom';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import history from './helpers/history';
import './App.css';
import ConnectedAuthenticatedApp from './components/AuthenticatedApp/AuthenticatedApp';
import ConnectedLogin from './components/Login/Login';

export function App({ jwt }) {
  return (
    <Router history={history}>
      <Switch>
        {jwt ? <ConnectedAuthenticatedApp /> : <ConnectedLogin />}
      </Switch>
    </Router>
  );
}

App.defaultProps = {
  jwt: null
};

App.propTypes = {
  jwt: PropTypes.string
};

const mapStateToProps = state => {
  return { jwt: state.userInfo.jwt };
};

const ConnectedApp = connect(mapStateToProps)(App);
export default ConnectedApp;
