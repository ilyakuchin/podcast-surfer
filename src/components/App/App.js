import React from "react";
import "./App.css";
import EpisodePlayer from "../EpisodePlayer/EpisodePlayer";

function App() {
  return (
    <div className="App">
      <EpisodePlayer
        pictureSrc="http://i1.sndcdn.com/avatars-000317750782-4p9exx-original.jpg"
        title="#117 Садись за стол, Братишка!"
        description="Вступление (0:35)
        Ищем евангелистов (16:33)
        Удаление интервью (21:23)
        ​​Google отказался от обучения нейросетей на бездомных темнокожих (25:04)
        Твиттер временно приостановил планы по удалению неактивных аккаунтов (28:10)
        VR (30:43)
        WASM-polyglot (34:06)
        Firefox Time Traveling (36:42)
        Боль с NestJS (39:31)
        Чему дальше учиться (55:39)
        
        Подать доклад на HolyJS https://holyjs-piter.ru/callforpapers/
        JSUnderhood https://twitter.com/jsunderhood
        ​​Google отказался от обучения нейросетей на бездомных темнокожих https://t.me/tprogerlive/5124
        Twitter delays deleting inactive accounts to decide how to respect dead users https://thenextweb.com/twitter/2019/11/27/twitter-delays-deleting-inactive-accounts-to-decide-how-to-respect-dead-users/
        Burning Lead https://burning-lead.timepad.ru/events/
        Сайт о туризме в США https://www.recreation.gov/
        How to create polyglot HTML/JS/WebAssembly module https://webassembly-security.com/polyglot-webassembly-module-html-js-wasm/
        Firefox Time Traveling https://firefox-replay.com/
        NestJS https://nestjs.com/
        Официальный сайт RxJS https://rxjs-dev.firebaseapp.com/
        Полезные демки с RxJS https://www.learnrxjs.io/
        Магазин с оборудованием https://www.musik-produktiv.com/
        Производство мебели https://www.instagram.com/levinart.spb/"
        audioSrc="http://feeds.soundcloud.com/stream/721823038-frontend_u-e117.m4a"
      ></EpisodePlayer>
    </div>
  );
}

export default App;
