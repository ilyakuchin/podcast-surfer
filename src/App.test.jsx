import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';
import toJson from 'enzyme-to-json';
import { App } from './App';

configure({ adapter: new Adapter() });

describe('Test App Component', () => {
  test('Should Render Correctly', () => {
    expect(toJson(shallow(<App username='username' />))).toMatchSnapshot();
  });
});
