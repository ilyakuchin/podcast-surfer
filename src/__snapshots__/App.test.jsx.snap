// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test App Component Should Render Correctly 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Switch>
    <Connect(Login) />
  </Switch>
</Router>
`;
